src/main/java/uk/gov/hmcts/reform/sendletter/data/LetterMapperFactory.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.sendletter.data;
+
+import org.springframework.jdbc.core.RowMapper;
+import uk.gov.hmcts.reform.sendletter.domain.LetterStatus;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+
+final class LetterMapperFactory {
+
+    public static final RowMapper<LetterStatus> LETTER_STATUS_MAPPER = new LetterStatusMapper();
+
+    private static final class LetterStatusMapper implements RowMapper<LetterStatus> {
+
+        @Override
+        public LetterStatus mapRow(ResultSet rs, int rowNumber) throws SQLException {
+            return new LetterStatus(
+                rs.getObject("id", UUID.class),
+                rs.getString("message_id"),
+                rs.getTimestamp("created_at").toLocalDateTime(),
+                rs.getTimestamp("sent_to_print_at").toLocalDateTime(),
+                rs.getTimestamp("printed_at").toLocalDateTime()
+            );
+        }
+    }
+
+    private LetterMapperFactory() {
+        // utility class constructor
+    }
+}
```

Comment:
why the whole wrapping class?

Reply:
figured there could be letter mapper e.g. for reporting. doesn't hurt to actually have this kind of approach